server/src/main/scala/za/co/absa/atum/server/api/RetrieveAwsSecret.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.server.api
+
+
+import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient
+import software.amazon.awssdk.services.secretsmanager.model.{GetSecretValueRequest, SecretsManagerException}
+import org.slf4j.{Logger, LoggerFactory}
+
+
+/**
+ * Class implement the functionality of retrieving secret keys from aws secret manger service
+ */
+class RetrieveAwsSecret (profileCredentials: String = "default") {
+  val secretsManagerClient: SecretsManagerClient = SecretsManagerClient.builder()
+    .region(Region.AF_SOUTH_1)
+    .credentialsProvider(ProfileCredentialsProvider.create(profileCredentials))
+    .build()
+
+  /**
+   * Function retrieves secret keys from aws
+   * @param secretName
+   * @return a sequence of string from aws secret service
+   */
+  def retrieveAwsSecret(secretName: String): Seq[String] = {
+    val logger: Logger = LoggerFactory.getLogger(getClass.getName)
+
+    try {
+      val request = GetSecretValueRequest.builder()
+        .secretId(secretName)
+        .build()
+
+      val response = secretsManagerClient.getSecretValue(request)
+      response.secretString.map(_.toString)
+    } catch {
+      case e: SecretsManagerException =>
+        logger.error(s"Error retrieving secret key: ${e.getMessage}")
+        e.getMessage.map(_.toString)
```

Comment:
This is certainly incorrect. You are again mapping over a `String`, therefore iterating over it's individual `Char` elements. Moreover you don't want to return this as a result of the method. How can a description of an exception `e.getMessage` be a valid return value of this method?

Reply:
Thanks for bringing that into my attention, I actually cooked that so it can silence the compiler. I have refactored it.